src/nvim/ops.c

```diff
@@ -3061,14 +3061,17 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
     if (gchar_cursor() == TAB) {
       /* Don't need to insert spaces when "p" on the last position of a
        * tab or "P" on the first position. */
+      int viscol = getviscol();
       if (dir == FORWARD
-          ? (int)curwin->w_cursor.coladd < curbuf->b_p_ts - 1
-          : curwin->w_cursor.coladd > 0)
-        coladvance_force(getviscol());
-      else
+          ? tabstop_padding(viscol, curbuf->b_p_ts, curbuf->b_p_vts_array) != 1
+          : curwin->w_cursor.coladd > 0) {
+        coladvance_force(viscol);
+      } else {
         curwin->w_cursor.coladd = 0;
-    } else if (curwin->w_cursor.coladd > 0 || gchar_cursor() == NUL)
-      coladvance_force(getviscol() + (dir == FORWARD));
+      }
+    } else if (curwin->w_cursor.coladd > 0 || gchar_cursor() == NUL) {
+        coladvance_force(getviscol() + (dir == FORWARD));
```

Comment:
indent

Reply:
Not positive what you mean here, what would be the correct indent?

Reply:
Neovim codestyle mandates 2-space indent. This has 4 space indent.

Reply:
If we're to automate that - where should it go? Into `make lint`? Seems like something you shouldn't care about during the review.

Reply:
https://github.com/neovim/neovim/blob/master/src/clint.py cannot format so `make lint` cannot not autofix lint errors. https://github.com/neovim/neovim/pull/13289 is WIP for clang-format.